org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -379,6 +407,26 @@ public Preferences setJavaHome(String javaHome) {
 		return this;
 	}
 
+	public Preferences setFormatterUrl(String formatterUrl) {
+		if (this.formatterUrl != null) {
+			File file = null;
+			try {
+				file = ResourceUtils.toFile(URIUtil.toURI(new URL(this.formatterUrl)));
```

Comment:
I remember that new URL() would actually go out to the internet and could block in the olden days. Not sure this still a problem.